### PR TITLE
Role should enable right repositories after attaching pools and autodiscover pools in case they not passed

### PR DESCRIPTION
**Pull request by KKoukiou**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1548083

### CODE REVIEW

**Comment by KKoukiou** [comment]
@mwperina @mykaul I followed the public available docs [here](https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.1/html/installation_guide/chap-installing_red_hat_enterprise_virtualization#Subscribing)  regarding which repos I should enable per version and per engine/host. However these are downstream repos. Do they fit as part of this role which is supposed to be used in upstream as well?

**Comment by mykaul** [comment]
I think the role should be usable for both RHEL/RHV (downstream) and EL/oVirt upstream - and the role should distinguish between them. I think a simple role would be RHEL -> RHV, otherwise -> oVirt.
(and then just add a
```
when: ansible_distribution == "RedHat"
```
statement)

**Comment by KKoukiou** [comment]
@mykaul I added the condition, thanks

File: README.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ Role Variables
 |--------------------------------------------|-----------------------|-------------------------------------------|
 | ovirt_repositories_ovirt_release_rpm       | UNDEF                 | URL of oVirt release package, which contains required repositories configuration. |
 | ovirt_repositories_use_subscription_manager| False                 | If true it will use repos from subscription manager and the value of ovirt_repositories_ovirt_release_rpm will be ignored |
+| ovirt_repositories_ovirt_version           | 4.2                   | oVirt release version. Will be used to enable the required repositories in case ovirt_repositories_use_subscription_manager is set. |
```
Comment (mwperina) [inline README.md]: I think we should mention possible values here: `master, 4.2,  4.1` Is that correct? But for subscription manager use case we need eliminate master, so we should probably document that

**Review by mwperina — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (KKoukiou) [reply README.md]: Sure, this variable is used only for subscription manager options and applicable for 4.1 and 4.2 BETA now. So I will add that supported options are [4.1, 4.2], ok?

Comment (mwperina) [reply README.md]: Ahh OK, I forgot that we specify ovirt_repositories_ovirt_release_rpm directly. So yes, for now let's limit to 4.1 and 4.2, in future we could use this version also for release rpm if release rpm is not specified.

Thanks

**Comment by KKoukiou** [comment]
@mwperina I added one extra commit which solves second issue https://bugzilla.redhat.com/show_bug.cgi?id=1548082 in the same PR since commits would have merge conflict anyway.

File: tasks/rh-subscription.yml [hunk 2]
```diff
@@ -1,10 +1,19 @@
 ---
-- name: Register and subscribe to multiple pools.
+- name: Register and subscribe to multiple pool IDs.
   redhat_subscription:
     state: present
     username: "{{ ovirt_repositories_rh_username | mandatory }}"
     password: "{{ ovirt_repositories_rh_password | mandatory }}"
     pool_ids: "{{ ovirt_repositories_pool_ids }}"
+  wnen:  ovirt_repositories_pool_ids is defined
```
Comment (machacekondra) [inline tasks/rh-subscription.yml]: s/wnen/when

File: tasks/rh-subscription.yml [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+
+- name: Use default subscription pool name for Red Hat Virtualization.
+  redhat_subscription:
+    state: present
+    username: "{{ ovirt_repositories_rh_username | mandatory }}"
+    password: "{{ ovirt_repositories_rh_password | mandatory }}"
+    pool: '^Red Hat Virtualization$'
+  wnen:  ovirt_repositories_pool_ids is not defined
```
Comment (machacekondra) [inline tasks/rh-subscription.yml]: s/wnen/when

File: tasks/rh-subscription.yml (hunk 2 above)
Comment (KKoukiou) [reply tasks/rh-subscription.yml]: ups, thanks

**Review by mwperina — APPROVED** [review]

**Comment by KKoukiou** [comment]
@machacekondra @mykaul Can you please CR ?

**Review by machacekondra — APPROVED** [review]

**Pull request merged**